library/Reporting/Web/Forms/TimeframeForm.php

```diff
@@ -41,34 +40,93 @@ public function hasBeenSubmitted(): bool
-            'required'                     => true,
-            'label'                        => $this->translate('End'),
-            'data-flatpickrDefaultHour'    => '23',
-            'data-flatpickrDefaultMinute'  => '59',
-            'data-flatpickrDefaultSeconds' => '59',
-            'placeholder'                  => $this->translate(
-                'Select a end date or provide a textual datetime description'
-            ),
-        ]);
+        if ($relativeStart === 'n') {
+            if (! $start instanceof DateTime) {
+                $start = $default;
+                $this->clearPopulatedValue('start');
+            }
+
+            $this->addElement(
+                new LocalDateTimeElement('start', [
+                    'required'    => true,
+                    'value'       => $start,
+                    'label'       => $this->translate('Start'),
+                    'description' => $this->translate('Specifies the start time of this timeframe')
+                ])
+            );
+        } else {
+            $this->addElement('text', 'start', [
+                'required'    => true,
+                'label'       => $this->translate('Start'),
+                'placeholder' => $this->translate('Provide a textual datetime description'),
+                'description' => $this->translate('Specifies the start time of this timeframe')
+            ]);
+        }
+
+        $default = new DateTime('23:59:59');
+        $end = $this->getPopulatedValue('end', $default);
+        if (! $end instanceof DateTime) {
+            $datetime = DateTime::createFromFormat(LocalDateTimeElement::FORMAT, $end);
+            if ($datetime) {
+                $end = $datetime;
+            }
+        }
+
+        $relativeEnd = $this->getPopulatedValue('relative-end', $end instanceof DateTime ? 'n' : 'y');
+        if ($relativeStart === 'y') {
+            $this->addElement('checkbox', 'relative-end', [
+                'required' => false,
+                'class'    => 'autosubmit',
+                'value'    => $relativeEnd,
+                'label'    => $this->translate('Relative End')
+            ]);
+        }
+
+        if ($relativeEnd === 'n') {
+            if (! $end instanceof DateTime) {
+                $end = $default;
+                $this->clearPopulatedValue('end');
+            }
+
+            $this->addElement(
+                new LocalDateTimeElement('end', [
+                    'required'    => true,
+                    'value'       => $end,
+                    'label'       => $this->translate('End'),
+                    'description' => $this->translate('Specifies the end time of this timeframe')
+                ])
+            );
+        } else {
+            $this->addElement('text', 'end', [
+                'required'    => true,
+                'label'       => $this->translate('End'),
+                'placeholder' => $this->translate('Provide a textual datetime description'),
```

Comment:
Maybe provide a usage example as the placeholder. (E.g. -30 days)

Same for the textual start time